examples/websockets/src/client.rs

```diff
@@ -0,0 +1,165 @@
+//! Based on tokio-tungstenite example websocket client, but with multiple
+//! concurrent websocket clients in one package
+//!
+//! This example will connect to a server specified in the argument in the specified
+//! number of threads, and then flood some test messages over websocket.
+//! This will also print whatever it gets into stdout.
+//!
+//! Note that this is not currently optimized for performance, especially around
+//! stdout mutex management. Rather it's intended to show an example of working with a
+//! websocket server.
+//!
+
+//boilerplate
+use clap::Parser;
+use std::borrow::Cow;
+use std::time::Instant;
+
+//we need tungstenite for websocket impl (same library as what axum is using)
+use tokio::task::JoinHandle;
+use tokio_tungstenite::tungstenite::protocol::frame::coding::CloseCode;
+use tokio_tungstenite::tungstenite::protocol::CloseFrame;
+use tokio_tungstenite::{connect_async, tungstenite::protocol::Message};
+use url::Url;
+//stream splitting
+use futures_util::{SinkExt, StreamExt};
+
+#[derive(Parser)]
+struct Args {
+    #[arg(short, long, default_value_t = String::from("ws://127.0.0.1:3000/ws"))]
+    url: String,
+    #[arg(short, long, default_value_t = 2)]
+    number_of_clients: usize,
+}
```

Comment:
I don't think we need all this argument parsing stuff, its not important to what the example is trying to show.

Could just hardcode the url and have `let number_of_clients = 10;` variable.

Reply:
good idea. got carried away.